evm/v0.5/testv2/Coordinator.test.ts

```diff
@@ -0,0 +1,909 @@
+        const tx = await initiateServiceAgreement(coordinator, serviceAgreement)
+        receipt = await tx.wait()
+      })
+
+      it('saves a service agreement struct from the parameters', async () => {
+        await h.checkServiceAgreementPresent(coordinator, serviceAgreement)
+      })
+
+      it('generates the SAID', async () => {
+        const ethSAID = await coordinator.getId(
+          h.encodeServiceAgreement(serviceAgreement),
+        )
+        assert.equal(ethSAID, sAID)
+      })
+
+      it('logs an event', async () => {
+        expect(
+          h.findEventIn(
+            receipt,
+            coordinator.interface.events.NewServiceAgreement,
+          ),
+        ).toBeDefined()
+      })
+
+      it('calls the aggregator with the SA info', async () => {
+        const event = h.findEventIn(
+          receipt,
+          coordinator.interface.events.NewServiceAgreement,
+        )
+        assert(event, 'event was expected')
+        const { said } = h.eventArgs(
+          h.findEventIn(
+            receipt,
+            coordinator.interface.events.NewServiceAgreement,
+          ),
+        )
+        assert.equal(said, sAID)
+      })
+    })
+
+    describe('with an invalid oracle signature', () => {
+      it('saves no service agreement struct, if signatures invalid', async () => {
+        const serviceAgreement = await newServiceAgreement({
+          oracles: [roles.oracleNode.address],
+          aggregator: emptyAggregator.address,
+        })
+        const sAID = h.generateSAID(serviceAgreement)
+        const badOracleSignature = await h.personalSign(sAID, roles.stranger)
+        const badRequestDigestAddr = h.recoverPersonalSignature(
+          sAID,
+          badOracleSignature,
+        )
+        assert.equal(roles.stranger.address, badRequestDigestAddr)
+        const conbinedSignatures = h.combineOracleSignatures([
+          badOracleSignature,
+        ])
+        await h.assertActionThrows(
+          coordinator.initiateServiceAgreement(
+            h.encodeServiceAgreement(serviceAgreement),
+            h.encodeOracleSignatures(conbinedSignatures),
+          ),
+        )
+        await h.checkServiceAgreementAbsent(coordinator, sAID)
+      })
+    })
+
+    describe('Validation of service agreement deadlines', () => {
+      it('Rejects a service agreement with an endAt date in the past', async () => {
+        const serviceAgreement = await newServiceAgreement({
+          endAt: 1,
+          aggregator: emptyAggregator.address,
+        })
+        const sAID = h.generateSAID(serviceAgreement)
+        await h.assertActionThrows(
+          initiateServiceAgreement(coordinator, serviceAgreement),
+        )
+        await h.checkServiceAgreementAbsent(coordinator, sAID)
+      })
+    })
+  })
+
+  describe('#oracleRequest', () => {
+    const to = '0x80e29acb842498fe6591f020bd82766dce619d43'
+    let agreement: h.ServiceAgreement, fHash: string, sAID: string
```

Comment:
For consistency's sake, I've been making multi-variable assignments into their own individual assignments. Would like to see this here